python/graphstorm/dataloading/dataset.py

```diff
@@ -787,3 +789,79 @@ def infer_idxs(self):
         """ Set of nodes to do inference.
         """
         return self._infer_idxs
+
+class GSDistillData(Dataset):
+    """ Dataset for distillation
+
+    Parameters
+    ----------
+    file_list : list of str
+        List of input files.
+    tokenizer : transformers.AutoTokenizer
+        HuggingFace Tokenizer.
+    max_seq_len : int
+        Maximum sequence length.
+    device : str
+        Device name.
+
+    """
+    def __init__(self, file_list, tokenizer, max_seq_len, device):
+        super().__init__()
+        self.file_list = file_list
+        self.tokenizer = tokenizer
+        self.max_seq_len = max_seq_len
+        self.device = device
+        self.token_id_inputs, self.labels = self.get_inputs()
+
+    def get_inputs(self):
+        """ Tokenize textual data."""
+        for i, file_name in enumerate(self.file_list):
+            if i == 0:
+                inputs = pd.read_parquet(file_name)
+            else:
+                inputs = pd.concat([inputs, pd.read_parquet(file_name)])
+
+        token_id_inputs = []
+        for i in range(len(inputs["textual_feats"])):
+            tokens = self.tokenizer.tokenize(inputs["textual_feats"][i])
+            tokens.insert(0, self.tokenizer.cls_token) # cls token for pooling
+            token_ids = self.tokenizer.convert_tokens_to_ids(tokens)
+            token_ids = token_ids[0:min(len(token_ids), self.max_seq_len)]
```

Comment:
Shouldn't tokenizer do this for us? Also if len(token_ids) < max_seq_len, do we need to do padding?

Reply:
yes these operations are done by tokenizer, and the padding is conducted in collate function.